### PR TITLE
Native linux support

### PR DESCRIPTION
**Pull request by freshtechtips**
## Overview

I know this sounds disappointing and actually I'm but I really don't have the time and it will take from me 6-12 months before I can touch the project so I decided that I at least should enable running the Linux example

So we can run and test it

I didn't do anything else
Just connected the Linux plugin to `pubspec.yaml`

This doesn't affect the other platforms and the only change that we can run the example on Linux

that's all

but the method channels will throw an Exception just like when using the library in a project that supports Linux

You can reject the PR as it doesn't change much, Linux is still unsupported, this is why I suggested to stick to the dart impl as I already implemented it before a few months ago

### CODE REVIEW

**Comment by natsuk4ze** [comment]
> discord

`@natsuk4ze.` you can find me in [pub channel on flutter](https://discord.gg/7PYfAhXx)

**Comment by natsuk4ze** [comment]
> I added the plugin interface for gal without new package or doing a breaking change

OK. 
I haven't done a thorough investigation of [dart-only-platform-implementations](https://docs.flutter.dev/packages-and-plugins/developing-packages#dart-only-platform-implementations) yet, but wouldn't this allow us to remove the `gal_linux` folder and put all your Linux implementations in a `linux` folder?

**Comment by freshtechtips** [comment]
> Two folders for linux is weird, so let's make it one.

Actually, it's one

But if you mean two folders

The other one for the example and this can't be deleted. Otherwise, we can't run the example on Linux

It is just for the example 

It won't affect the end developer who will use the library for linux

**Comment by freshtechtips** [comment]
All plugins have two folders for all platforms 

One for the example 
The other for the plugin itself

**Comment by natsuk4ze** [comment]
Then the linux_gal folder should be under linux.

**Comment by natsuk4ze** [comment]
One idea is to publish the linux support as your extension package and introduce it in the gal readme. This way you are free to write your code.

**Comment by natsuk4ze** [comment]
natsuk4ze. is my username

**Comment by freshtechtips** [comment]
> natsuk4ze. is my username 

Sent you a message there

File: example/.gitignore [hunk 1]
```diff
@@ -0,0 +1,43 @@
+# Miscellaneous
```
Comment (freshtechtips) [inline example/.gitignore:1]: This whole file is generated by `flutter create . --platforms=linux`

File: example/windows/flutter/ephemeral/.plugin_symlinks/gal [hunk 2]
```diff
@@ -0,0 +1 @@
+/home/freshtechtips/Desktop/gal/
```
Comment (freshtechtips) [inline example/windows/flutter/ephemeral/.plugin_symlinks/gal:1]: This should be deleted

File: pubspec.yaml [hunk 3]
```diff
@@ -3,10 +3,18 @@ description: Flutter Plugin for Saving Image/Video to the Photo Gallery | How to
...
 documentation: https://github.com/natsuk4ze/gal/wiki
 funding: 
  - https://github.com/sponsors/natsuk4ze
 
+platforms:
+  android:
+  ios:
+  linux:
+  macos:
+  windows:
```
Comment (freshtechtips) [inline pubspec.yaml:16]: Just to avoid `pubspec.yaml` from showing Linux as unsupported

**Pull request closed without merging**